pwnlib/fmtstr.py

```diff
@@ -709,7 +717,47 @@ def make_payload_dollar(data_offset, atoms, numbwritten=0, countersize=4):
         # perform write
         if padding:
             fmt += "%" + str(padding) + "c"
-        fmt += "%" + str(data_offset + idx) + "$" + SPECIFIER[atom.size]
+
+            if no_dollars:
+                data += pack(0)
```

Comment:
Because nullbytes often are string terminators, I would prefer some other value, like
```suggestion
                data += b'c' * context.bytes
```

Reply:
Also, it might be beneficial to use `b'c' * padding` on low values of padding (if padding < 4 on no_dollars=False and if padding < 4 + context.bytes on no_dollars=True).

Reply:
i see,  fo sure, i ve just put the first one that crossed my mind, i guess that ``context.bytes`` is the lenght of address based on the arch right?

Reply:
> Also, it might be beneficial to use `b'c' * padding` on low values of padding (if padding < 4 on no_dollars=False and if padding < 4 + context.bytes on no_dollars=True).

ahhh, i am not getting this part, what this about?

Reply:
> i see, fo sure, i ve just put the first one that crossed my mind, i guess that `context.bytes` is the lenght of address based on the arch right?

right, for example 8 on 64-bit and 4 on 32-bit

Reply:
> > Also, it might be beneficial to use `b'c' * padding` on low values of padding (if padding < 4 on no_dollars=False and if padding < 4 + context.bytes on no_dollars=True).
> 
> ahhh, i am not getting this part, what this about?

You can add `c` instead of `%1c`, `cc` instead of `%2c`, `ccc` instead of `%3c` and so on, if it is not longer than `%SOMETHINGc` to save off a few bytes.

Reply:
cool!  what about the other thing you told me bout, didn t really got it.

Reply:
sorry, tought the comments would self refresh, i ve wrote the last one w/o reading yours, i ll read now

Reply:
> > > Also, it might be beneficial to use `b'c' * padding` on low values of padding (if padding < 4 on no_dollars=False and if padding < 4 + context.bytes on no_dollars=True).
> > 
> > 
> > ahhh, i am not getting this part, what this about?
> 
> You can add `c` instead of `%1c`, `cc` instead of `%2c`, `ccc` instead of `%3c` and so on, if it is not longer than `%SOMETHINGc` to save off a few bytes.

make sense now, it is convenient just if the pad is less than 3,  if the pad is 3 it takes the same amount of chars
``ccc`` - ``%3c``, right? ill add a lil fix, check it back in a second.

Reply:
made it, have a look,  thx  :)